app/services/piece-access-level-service.js

```diff
@@ -62,4 +66,32 @@ export default class PieceAccessLevelService extends Service {
       previousPiece = await previousPiece.previousPiece;
     }
   }
+
+  /*
+   * This method is used to update the access levels of the pieces of a subcase's treatments. When a subcase has
+   * been updated to being confidential, all the pieces connected to it through agenda-item-treatments get the
+   * "Ministerraad" access level.
+   */
+  async updateDecisionsAccessLevelOfSubcase(subcase) {
+    await subcase.preEditOrSaveCheck();
+
+    if (!subcase.hasDirtyAttributes) {
+      return;
+    }
+
+    let { confidential: [, confidential] = [] } = subcase.changedAttributes();
+    if (!confidential) {
+      return;
+    }
+
+    const ministerraad = await this.store.findRecordByUri('access-level', CONSTANTS.ACCESS_LEVELS.MINISTERRAAD);
+
+    const treatments = await subcase.treatments;
+    for (const treatment of treatments.toArray()) {
+      const piece = await treatment.report;
```

Comment:
Maybe more efficient to use `this.store.query('document', ...)` with a filter on path to subcase id?
If we include `access-level` in the response we can even easily check, without additional network requests, whether the access level needs to be updated.